Source/Extensions/Blazorise.QRCode/QRCode.cs

```diff
@@ -63,30 +67,53 @@ public string GetAsBase64Image()
                 _ => throw new ArgumentOutOfRangeException()
             };
 
-            var darkColor = GetColorBytes( DarkColor );
-            var lightColor = GetColorBytes( LightColor );
+            var darkColor = GetColor( DarkColor );
+            var lightColor = GetColor( LightColor );
 
             using var data = Payload == null
                 ? generator.CreateQrCode( Value, eccLevel )
-                : generator.CreateQrCode( Payload, eccLevel );
+                : generator.CreateQrCode( Payload.ToString(), eccLevel );
 
-            var code = new PngByteQRCode( data );
-            var bytes = code.GetGraphic( PixelsPerModule, darkColor, lightColor, DrawQuietZones );
+            var code = new Base64QRCode( data );
 
-            const string pngData = "data:image/png;base64, ";
-            return pngData + Convert.ToBase64String( bytes );
+            if ( !string.IsNullOrEmpty( Icon ) )
+            {
+                using ( var image = GetImage( Icon ) )
+                {
+                    var imageType = GetImageType( IconImageType );
+
+                    return $"data:image/png;base64, {code.GetGraphic( PixelsPerModule, darkColor, lightColor, image, IconSizePercent, IconBorderWidth, DrawQuietZones, imageType )}";
+                }
+            }
+
+            return $"data:image/png;base64, {code.GetGraphic( PixelsPerModule, darkColor, lightColor, DrawQuietZones )}";
```

Comment:
On these lines we have : data:image/png;base64 is this relevant if it's a gif or jpeg? I did try setting a jpeg base64 and it did work fine visually.

Reply:
Good question. I will fix it.